pymc/distributions/timeseries.py

```diff
@@ -640,54 +651,120 @@ class GARCH11(distribution.Continuous):
+            batch_size = at.broadcast_shape(omega, alpha_1, beta_1, init_dist)
+        init_dist = change_dist_size(init_dist, batch_size)
 
-    def get_volatility(self, x):
-        x = x[:-1]
+        # Create OpFromGraph representing random draws form AR process
+        # Variables with underscore suffix are dummy inputs into the OpFromGraph
+        init_ = init_dist.type()
+        initial_vol_ = initial_vol.type()
+        omega_ = omega.type()
+        alpha_1_ = alpha_1.type()
+        beta_1_ = beta_1.type()
+        steps_ = steps.type()
 
-        def volatility_update(x, vol, w, a, b):
-            return at.sqrt(w + a * at.square(x) + b * at.square(vol))
+        noise_rng = aesara.shared(np.random.default_rng())
 
-        vol, _ = scan(
-            fn=volatility_update,
-            sequences=[x],
-            outputs_info=[self.initial_vol],
-            non_sequences=[self.omega, self.alpha_1, self.beta_1],
+        def step(*args):
+            prev_y, prev_sigma, omega, alpha_1, beta_1, rng = args
+            new_sigma = at.sqrt(
+                omega + alpha_1 * at.square(prev_y) + beta_1 * at.square(prev_sigma)
+            )
+            next_rng, new_y = Normal.dist(mu=0, sigma=new_sigma, rng=rng).owner.outputs
+            return (new_y, new_sigma), {rng: next_rng}
+
+        (y_t, _), innov_updates_ = aesara.scan(
+            fn=step,
+            outputs_info=[init_, initial_vol_],
+            non_sequences=[omega_, alpha_1_, beta_1_, noise_rng],
+            n_steps=steps_,
+            strict=True,
         )
-        return at.concatenate([[self.initial_vol], vol])
+        (noise_next_rng,) = tuple(innov_updates_.values())
+        garch11_ = at.concatenate([at.atleast_1d(init_), y_t.T], axis=-1)
 
-    def logp(self, x):
-        """
-        Calculate log-probability of GARCH(1, 1) distribution at specified value.
+        garch11_op = GARCH11RV(
+            inputs=[omega_, alpha_1_, beta_1_, initial_vol_, init_, steps_],
+            outputs=[noise_next_rng, garch11_],
+            ndim_supp=1,
+        )
 
-        Parameters
-        ----------
-        x: numeric
-            Value for which log-probability is calculated.
+        garch11 = garch11_op(omega, alpha_1, beta_1, initial_vol, init_dist, steps)
+        return garch11
 
-        Returns
-        -------
-        TensorVariable
-        """
-        vol = self.get_volatility(x)
-        return at.sum(Normal.dist(0.0, sigma=vol).logp(x))
 
-    def _distr_parameters_for_repr(self):
-        return ["omega", "alpha_1", "beta_1"]
+@_change_dist_size.register(GARCH11RV)
+def change_garch11_size(op, dist, new_size, expand=False):
+
+    if expand:
+        old_size = dist.shape[:-1]
+        new_size = tuple(new_size) + tuple(old_size)
+
+    return GARCH11.rv_op(
+        *dist.owner.inputs[:-1],
+        size=new_size,
+    )
+
+
+@_logprob.register(GARCH11RV)
+def garch11_logp(
```

Comment:
Question to @ricardoV94 is this necessary? I would imagine that rv_op already generate the appropriate rv and logp but i am getting non-implemented error without this.

Reply:
You need do set `aeppl_inline = True` in the `GARCH11RV` if you want to use Aeppl's derived logprob. 

You might still want to collapse the `logp` across the time axis, in which case you have to do something like this instead (and leave the default `aeppl_inline = False`: https://github.com/pymc-devs/pymc/blob/4954fd4fead9608f3e1bff0f2e6ce33ca6170d1b/pymc/distributions/timeseries.py#L231-L242

Reply:
thanks! I will try it.

Reply:
Have you checked if Aeppl can figure this one out? Specially with the transposition and concatenate operation? It might not...